Docs update list widget

rlahoda opened the pull request:
**Summary**

When setting up Gatsbyjs with netlify cms I was having errors with the `config.yml` because I was following the documentation for the "list" widget. Specifically the documentation lists that when the `fields` option is used it should be delineated:
```
fields:
    - {...}
    - {...}
```
However the boilerplate code actually is set up:
```
fields:
    [
      {...},
      {...}
    ]
```

So this is a tweak to the documentation reflecting this change.

**Test plan**

This is just updating an .md file so no testing should be necessary

**A picture of a cute animal (not mandatory but encouraged)**
Our pup, Ember with her favorite chew toy.
![ember](https://user-images.githubusercontent.com/18124644/73141490-7ef98880-4052-11ea-9ba4-536ea02198a1.jpg)

erezrokah commented:
Hi @rlahoda, thank you for submitting this change. Can you share the errors you were having and the config file that generated them?
I've created a config file based on the current documentation here: https://github.com/erezrokah/netlify-cms-reproductions/blob/netlify_cms/pr_3142/public/config.yml and it seems to work.

Also I think you're missing some indentations:
Official docs:
![image](https://user-images.githubusercontent.com/26760571/73227368-357b6d00-4128-11ea-8727-5beb7d3f7c36.png)

With your change (https://deploy-preview-3142--netlify-cms-www.netlify.com/docs/widgets/#list):
![image](https://user-images.githubusercontent.com/26760571/73227388-49bf6a00-4128-11ea-98ad-374569825124.png)

rlahoda commented:
Hi @erezrokah, Thanks for taking a look at this. Do you have any idea why they might be not indenting correctly? I looked at the markdown preview even on github and they do so I'm not sure if I'm missing something. I did notice that my code formatter automatically took out the leading spaces on all of the lines so maybe that's part of it? 

I'll attach the other info soon. Thanks!

rlahoda commented:
Here's the error I was getting. As I switched over from the one format to the other when defining the `fields`:
```
Error loading the CMS configuration
Config Errors:

YAMLException: missed comma between flow collection entries at line 48, column 18:
                field: {label: Image, name: image, wi ... 
                     ^

Check your config.yml file.
```

As I changed the format of `fields:` the same error would pop up going down the page until I got through all of them.

Strangely, I did notice that the issue only seemed to happen when `fields:` was a few levels deep. On line 21 in both files, it's using the `- {...}` syntax and working fine, but on line 40 it would error unless it used the `[ {...} ]` syntax. 

Here's the yml file as well from that same commit. Github wouldn't let me upload it as a .yml file so i changed the extension to .txt
[config.txt](https://github.com/netlify/netlify-cms/files/4120561/config.txt)

Here's the same file after I fixed it and got it working. There might be some minor changes as I was tweaking other stuff while trying to figure out the issue.
[config-fixed.txt](https://github.com/netlify/netlify-cms/files/4120575/config-fixed.txt)

erquhart commented:
@rlahoda yaml arrays don’t require explicit brackets, so this change shouldn’t be necessary - if you open an issue we can definitely help troubleshoot 👍👍

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

artokun commented:
I'm having this exact same issue, seems that nested `fields` is not parsing correctly unless you use the bracket notation. 

This is the only way i could get it to work without the error
```yaml
backend:
name: git-gateway
branch: master

media_folder: src/images
public_folder: /images

collections:
  - name: aboutUs
    label: About Us Page
    fields:
      - {
          name: ourTeam,
          label: Our Team,
          widget: "list",
          fields:
            [
              { label: Name, name: name, widget: string },
              { label: Title, name: title, widget: string },
              { label: Bio, name: bio, widget: text },
              { label: Avatar, name: avatar, widget: image },
            ],
        }
```
